### PR TITLE
Don't assume `schema.properties` exists. #44

### PR DESCRIPTION
**Pull request by chmac**
This is **completely untested**.

I'm not able to easily test this on our system, but as best I can tell without testing, I guess it'll work. :-)

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -67,10 +67,14 @@ export function getDefaultFormState(schema) {
...
-      return acc;
-    }, {});
+    if (schema.properties) {
+      return Object.keys(schema.properties).reduce((acc, key) => {
+        acc[key] = getDefaultFormState(schema.properties[key]);
+        return acc;
+      }, {});
+    } else {
+      return {}
+    }
```
Comment (n1k0) [inline src/utils.js]: We could actually simply default the passed `schema.properties` value in `Object.keys`:

```js
return Object.keys(schema.properties || {}).reduce((acc, key) => {
```

**Comment by n1k0** [comment]
Thank you!

I'm expecting a unit test for this change; if you struggle with writing one I can take care of it :)

File: src/utils.js [hunk 2]
```diff
@@ -67,10 +67,14 @@ export function getDefaultFormState(schema) {
...
-      acc[key] = getDefaultFormState(schema.properties[key]);
-      return acc;
-    }, {});
+    if (schema.properties) {
+      return Object.keys(schema.properties).reduce((acc, key) => {
+        acc[key] = getDefaultFormState(schema.properties[key]);
+        return acc;
+      }, {});
+    } else {
+      return {}
```
Comment (n1k0) [inline src/utils.js]: Nit: Missing semicolon (hence build failure). You can run `npm run lint` before pushing to ensure it doesn't break :)

File: src/utils.js (hunk 1 above)
Comment (chmac) [inline src/utils.js]: Smart, that's a much better approach. I'll kill this PR and create a new one with only that change, keep the git history neater.

**Comment by chmac** [comment]
I'll have a go at a unit test also! :-)

**Pull request closed without merging**